src/components/timecard/time-input-errors/TimeInputErrors.jsx

```diff
@@ -0,0 +1,123 @@
+import { PropTypes } from 'prop-types';
+import { useApplicationContext } from '../../../context/ApplicationContext';
+import {
+  formatLongDate,
+  formatTime,
+  getTimePeriodTypesMap,
+} from '../../../utils/time-entry-utils/timeEntryUtils';
+
+const TimeInputErrors = ({ clashingProperty, clashes }) => {
+  const { timePeriodTypes } = useApplicationContext();
+  const timePeriodTypesMap = getTimePeriodTypesMap(timePeriodTypes);
+
+  const displayClashingProperty = () => {
+    if (clashingProperty === 'startTime') {
+      return <p>Your start time must not overlap with another time period</p>;
+    }
+
+    if (clashingProperty === 'endTime') {
+      return <p>Your finish time must not overlap with another time period</p>;
+    }
+
+    return (
+      <p>
+        Your start and finish times must not overlap with another time period
+      </p>
+    );
+  };
+
+  const displaySingleTimeClash = () => {
+    if (clashes.length === 0) {
+      throw new Error(
+        'The time clashes data did not contain at least one time clash.'
+      );
+    }
+    const clash = clashes[0];
+    let text;
+
+    const timePeriodType = timePeriodTypesMap[clash.timePeriodTypeId];
+
+    if (timePeriodType === 'Shift') {
+      text = `You are already assigned to work from ${formatStartDateTime(
+        clash
+      )} to ${formatTime(clash.endTime)} on ${formatLongDate(clash.endTime)}`;
+    } else {
+      text = `You are already assigned a ${timePeriodType.toLowerCase()} on ${formatLongDate(
+        clash.startTime
+      )}`;
+    }
+
+    return <p>{text}</p>;
+  };
+
+  const displayMultipleTimeClashes = () => {
+    sortClashes(clashes);
+
+    const times = (
+      <div>
+        <p>You are already assigned to the following time periods:</p>
+        <ul>
+          {clashes.map((clash, index) => (
+            <li key={index} data-testid="test">
+              {timePeriodClashToText(clash)}
+            </li>
+          ))}
+        </ul>
+      </div>
+    );
+    return times;
+  };
+
+  const sortClashes = (clashes) => {
```

Comment:
Consider renaming this (e.g. `sortByStartTime`) to let future code maintainers understand its purpose, and moving to a utility if it could be reused elsewhere in the application

Reply:
I've renamed it, I think it might be too specific to be reused elsewhere?

Reply:
Looks good. Ok - just a thought :)